Content.Shared/GameObjects/Components/Chemistry/SharedSolutionContainerComponent.cs

```diff
@@ -1,115 +1,228 @@
 
-                _substanceColor = value;
-                Dirty();
+            var primaryReagent = Solution.GetPrimaryReagentId();
+            if (!prototypeManager.TryIndex(primaryReagent, out ReagentPrototype proto))
+            {
+                Logger.Error($"{nameof(SharedSolutionContainerComponent)} could not find the prototype associated with {primaryReagent}.");
+                return;
             }
+
+            var colorHex = Color.ToHexNoAlpha(); //TODO: If the chem has a dark color, the examine text becomes black on a black background, which is unreadable.
+            var messageString = "It contains a [color={0}]{1}[/color] " + (ReagentList.Count == 1 ? "chemical." : "mixture of chemicals.");
+
+            message.AddMarkup(Loc.GetString(messageString, colorHex, Loc.GetString(proto.PhysicalDescription)));
         }
 
-        /// <summary>
-        ///     The current capabilities of this container (is the top open to pour? can I inject it into another object?).
-        /// </summary>
-        [ViewVariables(VVAccess.ReadWrite)]
-        public SolutionContainerCaps Capabilities { get; set; }
+        private void UpdateAppearance()
+        {
+            if (!Owner.TryGetComponent<SharedAppearanceComponent>(out var appearance))
+                return;
 
-        public abstract bool CanAddSolution(Solution solution);
+            appearance.SetData(SolutionContainerVisuals.VisualState, GetVisualState());
+        }
 
-        public abstract bool TryAddSolution(Solution solution, bool skipReactionCheck = false, bool skipColor = false);
+        private SolutionContainerVisualState GetVisualState()
+        {
+            var filledVolumeFraction = CurrentVolume.Float() / MaxVolume.Float();
 
-        public abstract bool TryRemoveReagent(string reagentId, ReagentUnit quantity);
+            return new SolutionContainerVisualState(Color, filledVolumeFraction);
+        }
 
-        /// <inheritdoc />
         public override ComponentState GetComponentState()
         {
             return new SolutionContainerComponentState(Solution);
         }
 
         public override void HandleComponentState(ComponentState? curState, ComponentState? nextState)
         {
-            base.HandleComponentState(curState, nextState);
-
-            if (curState is not SolutionContainerComponentState state)
-            {
+            if (curState is not SolutionContainerComponentState containerState)
                 return;
-            }
 
-            _solution = state.Solution;
+            Solution = containerState.Solution;
+        }
+    }
+
+    [Serializable, NetSerializable]
+    public enum SolutionContainerVisuals
+    {
+        VisualState
+    }
+
+    [Serializable, NetSerializable]
+    public class SolutionContainerVisualState
+    {
+        public readonly Color Color;
+
+        public readonly float FilledVolumeFraction;
+
+        public SolutionContainerVisualState(Color color, float filledVolumeFraction)
```

Comment:
If you _really_ wanted to min-max bandwidth you could use a byte instead for FilledVolumeFraction.